src/mulang.js

```diff
@@ -1,52 +1,6 @@
-
-//-----------
-// Getters --
-//-----------
-
-function get(key, object) {
-  if (object === undefined) throw new Error("undefined object, when asking for key " + key)
-  if (key === undefined) throw new Error("undefined key")
-  return object[key]
-}
-
-function getJust(key, object) {
-  var value = get(key, object);
-  if (value === undefined) {
-    throw new Error("value not present");
-  }
-  return value;
-}
-
-function getWith(f, key, object) {
-  return f(getJust(key, object));
-}
-
-function getArrayWith(f, key, object) {
-  return getWith(it => parseArray(f, it), key, object);
-}
-
-function getString(key, object) {
-  return getStringWith((x) => x, key, object);
-}
-
-function getStringWith(f, key, object) {
-  function ensureString(s) {
-    if (typeof s !== 'string') {
-      throw new Error("value is not an string");
-    }
-    return f(s);
-  }
-  return getWith(ensureString, key, object);
-}
-
-function getExpression(key, object) {
-  return getWith(parseExpression, key, object);
-}
-
-function getBody(key, object) {
-  return getWith(parseBody, key, object);
-}
-
+// TODO:
+// - Interactive programs (N_DefInteractiveProgram)
+// - Foreach (N_StmtForeach) and ranges (N_ExprRange)
```

Comment:
Please return `{ "tag" :  "Other" }` on unsupported elements.

Reply:
Just FYI, this is the - yet unmerged - spec of the foreach in mulang: https://github.com/mumuki/mulang/pull/98/files